tensorflow/core/kernels/mkl_pooling_ops_common.cc

```diff
@@ -106,20 +121,29 @@ void MklPoolingFwdPrimitive<T>::Execute(const T* src_data, T* dst_data,
   context_.src_mem->set_data_handle(
       static_cast<void*>(const_cast<T*>(src_data)));
   context_.dst_mem->set_data_handle(static_cast<void*>(dst_data));
-  if (context_.alg_kind == pooling_max &&
+  if (context_.alg_kind == ALGORITHM::pooling_max &&
       context_.prop_kind ==
-          prop_kind::forward_training) {  // max pooling must have ws
+          prop_kind::forward_training) {  // Max pooling must have workspace.
     DCHECK(ws_data != nullptr);
     context_.ws_mem->set_data_handle(ws_data);
   }
+
+#ifdef ENABLE_MKLDNN_V1
```

Comment:
Seems like this (lines 132-138) is used in many ops. Maybe refactor it into `mkl_util` or something? Please also use this for `bwd_primitives`.

Reply:
This hasn't been addressed yet. (If it's not worth doing, please explain why. Otherwise, please add a TODO or make modifications.)

Reply:
@penpornk This is what I was looking into if at all it is possible to do this in a simple way. I think you might have misunderstood mkl op context_ to be a generic class as such as the Eigen OpKernelContext. unfortunately, the mkl op context_ is specific to each op and specific to fwd and bwd prop within each of those op. Going through the MklPrimitive also did not seem like a simple fix and needs a decent redesign. Opened an internal ticket for this to be redesigned for east of use in future. So it is worth doing, but Not easily doable without a significant redesign.

Reply:
But the primitives and streams in each context are of the same type, right? In here (and `mkl_conv_grad_filter_ops.cc` and `mkl_softmax_op.cc`) `fwd_primitives` `bwd_primitives` are always of type `mkldnn:primitive` and `fwd_stream` is of type `mkldnn::stream`. I'm thinking of something along the line of 
```c++
// In mkl_utils.h or some other common file.
#ifdef ENABLE_MKLDNN_V1
void execute_primitives(std::vector<mkldnn::primitive>& primitives, 
                        std::shared_ptr<stream> stream, 
                        std::vector<std::unordered_map<int, memory>>& net_args) {
  DCHECK_EQ(primitives.size(), net_args.size();
  for (size_t i = 0; i < primitives.size(); ++i) {
    primitives.at(i).execute(*stream, net_args.at(i));
  }
}
#endif  // ENABLE_MKLDNN_V1

// In this file.
#ifdef ENABLE_MKLDNN_V1
  execute_primitives(context_.fwd_primitives, context_.fwd_stream, context_.net_args);
#else 
  context_.fwd_stream->submit(context_.fwd_primitives);
#endif  // ENABLE_MKLDNN_V1
```

I think this is trivial enough to do in this PR. But if you strongly prefer doing it in a later PR, I'm okay with that too.